### ISSUE TITLE
Setup from 2.4.4 broken in 3.0.0

### ISSUE THREAD

**Issue by feuGeneA** [sphinx-doc/sphinx#7429]
I've had an automated build running well for about a year, always getting the latest sphinx updates.  The 3.0.0 update broke my build.

The output I'm seeing is a _ton_ of repeated messages exactly like this:

```
/usr/local/lib/python3.7/site-packages/sphinx/util/inspect.py:546: RemovedInSphinx40Warning: sphinx.util.inspect.Signature() is deprecated
  RemovedInSphinx40Warning)
```

Followed by a final, fatal message:

```
Warning, treated as error:
/usr/local/lib/python3.7/site-packages/zero_ex/contract_wrappers/exchange/__init__.py:docstring of zero_ex.contract_wrappers.exchange.LibOrderOrder:1:duplicate object description of zero_ex.contract_wrappers.exchange.LibOrderOrder, other instance in index, use :noindex: for one of them


Exited with code exit status 1
```

I updated all the packages in my local environment and reproduced the problem there as well.

I've attached the build log: [build_112627_step_106_container_0.txt](https://github.com/sphinx-doc/sphinx/files/4441932/build_112627_step_106_container_0.txt)

And you can see it in the CI context in case that helps:: https://circleci.com/gh/0xProject/0x-monorepo/112592?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

The sphinx setup for this package is rooted at at https://github.com/0xProject/0x-monorepo/blob/development/python-packages/contract_wrappers/setup.py#L263

However, the code referred to by that fatal error can't be found in that repository, because it's generated during the build process.  Here's the generated output from my machine, where it's failing the same way: 
[__init__.py.zip](https://github.com/sphinx-doc/sphinx/files/4441979/__init__.py.zip)

**Environment info**
- OS: Ubuntu on my build server, Debian on my workstation
- Python version: 3.7.2 on my workstation, 3.7.something on my build server
- Sphinx version: 3.0.0
- Sphinx extensions: sphinx-autodoc-typehints

**Comment by gpotter2** [comment]
The first issue is pretty easy: https://github.com/agronholm/sphinx-autodoc-typehints/blob/a25fa05084b4cee7e9e733603b79e68986a0ffee/sphinx_autodoc_typehints.py#L8

You are still using `sphinx-autodoc-typehints` which hasn't been updated to get rid of the deprecation. However you shouldn't need the extension anymore: 3.0.0 has added official support for typehints (it was already partially there since 2.4 if i recall, but a sphinx dev will know more than i do..)

The second part looks like a duplicate of https://github.com/sphinx-doc/sphinx/issues/7427.

**Comment by tk0miya** [comment]
Yes, the first one is already filed as https://github.com/agronholm/sphinx-autodoc-typehints/issues/133 . Please follow the issue.

Another one is a warning, not an error. I guess you're using `-W` option. And it promotes a warning to an error. And please check your document without `-W` option. I guess your document has duplicated entry about `zero_ex.contract_wrappers.exchange.LibOrderOrder`. Please check it.

**Comment by ariso353** [comment]
I am having similar issues.  All of our sites that were built with 2.4.4 are now broken with 3.0.0.  I'm also getting BS errors for indentation when there is none.

**Comment by feuGeneA** [comment]
@tk0miya yes, you're right, I am treating warnings as errors, thank you for pointing that out.

The problem was that I had the class' docstring specified in my index.rst _and also_ in the definition of the class itself.  I was using this as a way to append one docstring (the index.rst one) onto the end of the other (the class def one) for my rendered docs.  I've now changed to using an `autodoc-process-docstring` hook in my conf.py, and the warning is resolved.

I also ditched `sphinx-autodoc-typehints`, cleaning up my warnings.

Thank you for the support.
